assets/js/job-editor/JobEditorMain.tsx

```diff
@@ -0,0 +1,56 @@
+import React from "react"
+import JobEditor from "./JobEditor"
+import type { WithActionProps } from "../react/lib/with-props";
+import { sortMetadata } from "../metadata-loader/metadata";
+import { useWorkflowStore } from "../workflow-editor/store";
+import pDebounce from "p-debounce";
+import { EDITOR_DEBOUNCE_MS } from "../common";
+import type { Lightning } from "#/workflow-diagram/types";
+
+interface JobEditorComponentProps {
+  job_id: string,
+  adaptor: string,
+  source: string,
+  disabled: boolean,
+  disabled_message: string,
+}
+
+export const JobEditorMain: WithActionProps<JobEditorComponentProps> = (props) => {
+  const [metadata, setMetadata] = React.useState<false | object>(false);
+  const [source, setSource] = React.useState('');
+
+  // use workflowstore here
+  const { change, getById } = useWorkflowStore();
+
+  // debounce editor content update
+  const _debouncedPushChange = pDebounce((content: string) => {
+    change({ jobs: [{ id: props.job_id, body: content }] })
+  }, EDITOR_DEBOUNCE_MS);
+
+  // init hook - getting metadata
+  React.useEffect(() => {
+    props.handleEvent('metadata_ready', (payload) => {
+      setMetadata(payload as object);
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+      const sortedMetadata = sortMetadata(payload);
+      setMetadata(sortedMetadata as object);
+    })
+    props.pushEventTo('request_metadata', {});
+  }, [props])
+
+  React.useEffect(() => {
+    const foundJob = getById<Lightning.Job>(props.job_id);
```

Comment:
Is this behaviour the same as production, or have we changed? I guess props.source is the placeholder text?

Reply:
After the store is connected. JobEditor gets it's source from the store. Hence this is for when `job_id` changes. It searches for the source of a job in the store.